src/macros.jl

```diff
@@ -192,3 +192,19 @@ macro trixi_testset(name, expr)
         nothing
     end
 end
+
+"""
+    @test_allocations(mod, semi, sol, allocs)
+
+Test that the memory allocations of `mod.rhs!` are below `allocs`
+(e.g., from type instabilities), where `mod` is a module containing
+a method `rhs!(du, u, semi, t)`.
+"""
+macro test_allocations(mod, semi, sol, allocs)
+    quote
+        t = $(esc(sol)).t[end]
+        u = $(esc(sol)).u[end]
+        du = similar(u)
+        @test (@allocated $(esc(mod)).rhs!(du, u, $(esc(semi)), t)) < $(esc(allocs))
+    end
+end
```

Comment:
Wouldn't it be better to pass the `rhs!` we should use directly as an argument to the macro?

Reply:
We could also do that. What exactly would be the benefit?

Reply:
Ah, I forgot we have a `rhs_parabolic!`. I can adapt that :+1